### PR TITLE
Clean up query test

### PR DESCRIPTION
**Pull request by dcbrandao**
- Add absl-py for doing parameterized testing
- Add helpers to generate SelectQueries for select queries against
  UnittestModel and includes queries against ChildTestModel
- Use public SelectQuery methods (sql, parameters, and types) instead of
  internal methods (_where, _limit, etc)
- Updated setup.py so that ```python setup.py test``` still works

### CODE REVIEW

File: spanner_orm/tests/query_test.py [hunk 1]
```diff
@@ -55,124 +57,120 @@ def test_count(self, mock_db):
...
   def test_query_limit(self):
-    key, value = ('limit0', 2)
-    query_ = query.SelectQuery(models.UnittestModel, [condition.limit(value)])
+    key, value = 'limit0', 2
+    select_query = self.select(condition.limit(value))
 
-    self.assertTrue(query_.sql().endswith(' LIMIT @{}'.format(key)))
-    self.assertEqual(query_.parameters(), {key: value})
-    self.assertEqual(query_.types(), {key: field.Integer.grpc_type()})
+    self.assertTrue(select_query.sql().endswith(' LIMIT @{}'.format(key)))
```
Comment (akefeli) [inline spanner_orm/tests/query_test.py]: you can use `self.assertEndsWith`

File: spanner_orm/tests/query_test.py [hunk 2]
```diff
@@ -55,124 +57,120 @@ def test_count(self, mock_db):
...
 
   def test_query_limit_offset(self):
     limit_key, limit = 'limit0', 2
     offset_key, offset = 'offset0', 5
-    query_ = query.SelectQuery(models.UnittestModel,
-                               [condition.limit(limit, offset=offset)])
+    select_query = self.select(condition.limit(limit, offset=offset))
 
-    self.assertTrue(query_.sql().endswith(' LIMIT @{} OFFSET @{}'.format(
+    self.assertTrue(select_query.sql().endswith(' LIMIT @{} OFFSET @{}'.format(
```
Comment (akefeli) [inline spanner_orm/tests/query_test.py]: assertEndsWith

File: spanner_orm/tests/query_test.py [hunk 3]
```diff
@@ -55,124 +57,120 @@ def test_count(self, mock_db):
...
     order = ('int_', condition.OrderType.DESC)
-    query_ = query.SelectQuery(models.UnittestModel,
-                               [condition.order_by(order)])
+    select_query = self.select(condition.order_by(order))
 
-    self.assertTrue(query_.sql().endswith(' ORDER BY table.int_ DESC'))
-    self.assertEqual(query_.parameters(), {})
-    self.assertEqual(query_.types(), {})
+    self.assertTrue(select_query.sql().endswith(' ORDER BY table.int_ DESC'))
+    self.assertEqual(select_query.parameters(), {})
```
Comment (akefeli) [inline spanner_orm/tests/query_test.py]: assertEmpty

File: spanner_orm/tests/query_test.py [hunk 4]
```diff
@@ -223,32 +217,24 @@ def test_includes_subcondition_result(self):
...
-      _ = query_.process_results(rows)
+      _ = select_query.process_results(rows)
 
   def test_includes_error_on_invalid_relation(self):
     with self.assertRaises(AssertionError):
-      query.SelectQuery(models.ChildTestModel,
-                        [condition.includes('bad_relation')])
+      self.includes('bad_relation')
 
   def test_includes_error_on_invalid_subconditions(self):
```
Comment (akefeli) [inline spanner_orm/tests/query_test.py]: how about parameterizing these?

File: spanner_orm/tests/query_test.py (hunk 1 above)
Comment (dcbrandao) [reply spanner_orm/tests/query_test.py]: 👍

**Pull request merged**